### ISSUE TITLE
Difference to Julia?

### ISSUE THREAD

**Issue by Wikunia** [FBMachine/bon#3]
Hi, 
I just had a small look here without installing it yet. Wanted to ask whether there are some benefits compared to Julia:
https://julialang.org
It uses jit as well and has a nice package support. As far as I know it isn't possible to precompile it yet though. 

Thanks!

**Comment by daniel-p-gonzalez** [comment]
Hi,
Thanks for taking a look. Here are some differences:

Unlike Julia's dynamic type system, Bon has a static type system which is probably best described as OCaml with typeclasses. That's not to say that Bon can do something that Julia can't here - there are pattern matching packages for Julia using macros, and function overloading with support for multiple dispatch can be done through means other than typeclasses. However, Bon's type system aims to have a simpler core that is easier to reason about, and which catches more errors at compile time.

Julia is aimed at the numerical computing community, while Bon aims to be more general purpose. So while Bon will include e.g. a tensor type in the stdlib for numerical computing, it will also include modules that help you build web servers, games, and GUI applications.

In short, if you're looking to replace Matlab with a more expressive and performant language, Julia is a great choice. If you're looking to replace Python with a language that is more performant and has a powerful static type system, you should keep an eye on Bon.
